superset-frontend/src/explore/components/ControlPanelsContainer.tsx

```diff
@@ -269,6 +270,29 @@ export const ControlPanelsContainer = (props: ControlPanelsContainerProps) => {
 
   const containerRef = useRef<HTMLDivElement>(null);
 
+  const controlsTransferred = useSelector<ExplorePageState, string[]>(
+    state => state.explore.controlsTransferred,
+  );
+
+  useEffect(() => {
+    if (props.chart.chartStatus === 'success') {
+      controlsTransferred.forEach(controlName => {
+        const alteredControls = ensureIsArray(
+          props.controls[controlName].value,
+        ).map(value => {
+          if (
+            typeof value === 'object' &&
+            isDefined(value) &&
+            'datasourceWarning' in value
```

Comment:
Minor. Do you think we can have Typescript helping here or maybe setting it in a constant? I am not very fond of using a plain string for `datasourceWarning`

Reply:
Unfortunately the typing of that area of code is very poor and `value`'s type is basically any object, so addind `datasourceWarning` to typing here would require a lot of restructuring which is not a focus of this PR.
However, I think `'fieldName' in objectName` is a rather common pattern, we use it in many places in the code, including typescript files

Reply:
To be precise, there are 78 occurences across 27 typescript files of this pattern 🙂